internal/descriptor/types/ocmidentity/ocmidentity.go

```diff
@@ -0,0 +1,47 @@
+package ocmidentity
+
+import (
+	"errors"
+	"fmt"
+)
+
+var ErrValueNotProvided = errors.New("value not provided")
+
+// Component uniquely identifies an OCM Component.
+// See: https://ocm.software/docs/overview/important-terms/#component-identity
+type Component struct {
+	componentName    string
+	componentVersion string
+}
+
+// MustNew is a convenience constructor that panics if name or version are not provided.
+func MustNew(name, version string) *Component {
+	c, err := New(name, version)
+	if err != nil {
+		panic(err)
+	}
+	return c
+}
+
+// New is a constructor that ensures that both name and version are provided.
+func New(name, version string) (*Component, error) {
```

Comment:
go have naming convesion to initilze struct, normally should be `New[struct name]`, so here `NewComponent` is better. Actually I also second @lindnerby suggestions, rename it to `ComponentId` or maybe better name

Reply:
Changed to `NewComponentId`